post 서비스를 위한 DTO 객체 생성

lecileJJL opened the pull request:
### PR Type
<!— Please check the one that applies to this PR using "x". —>

- [o] 버그수정(Bugfix)
- [ ] 기능개발(Feature)
- [ ] 코드 스타일 변경(Code style update) (formatting, local variables)
- [o] 리팩토링 (no functional changes, no api changes)
- [ ] 빌드 관련 변경
- [ ] 문서 내용 변경
- [ ] Other… Please describe:

### 요약(Summary)
Post에 사용될 데이터 DTO를 업로드 , 수정 중에 있습니다.

### 상세 내용(Describe your changes)
기존 inner Class로 작성 된 PostDTO.java 파일 삭제 후 CreatePostReqeustDTO.java , ListPostRequstDTO.java, ListPostResponseDTO.java , PostDTO.java ,ResponsePostDTO.java 로 나눠 작성 했습니다. 

PostDTO.java : 단일 post의 내용을 비밀번호를 제외 하고 담은 객체
ResponsePostDTO.java : 단일 post를 반환 하는 객체
CreatePostReqeustDTO.java PostDTO 객체 와 password를 reqeust에 이용
ListPostRequstDTO.java : 페이지 수 페이지 사이즈 값 포함 페이지 요청 시 이용
ListPostResponseDTO.java : 요청받은 페이지 만큼 총 post의 개수와 함꼐 반환


-- 기존 불필요한 중복 코드 제거 및 PostDTO.java 객체를 참조 하여 이외 객체를 정의하였습니다.
ps database에 nickname 컬럼이 없는 것을 발견 해 추가 하였습니다.


### Issue Number or Link

lecileJJL committed:
create postDTO.java

postDTO.java 파일을 새로 작성 했습니다. 확장성을 생각해 inner class 형식으로 작성했습니다.

sesang06 commented:
좀더 PR의 타이틀과 디스크립션을 더 성의있게 써 주세요. 어떤 목적을 하는 PR인지 알기 어렵습니다.

sesang06 commented on src/main/java/com/example/PetnityServer/data/dto/PostDTO.java:
```diff
@@ -0,0 +1,54 @@
+import lombok.Getter;
+
+public class PostDTO {
+    public static class Request {
```
왜 static으로 작성하셨는지요?

lecileJJL replied:
static으로 선언 하지 않을 경우 inner class가 외부 참조를 하게 되어 메모리 누수가 발생하기 때문입니다.  그 원인으로 static없이 선언하고 빌드하게 될 수 컴파일 과정에서 경고가 발생합니다

sesang06 replied:
설명 감사합니다.

sesang06 commented on src/main/java/com/example/PetnityServer/data/dto/PostDTO.java:
```diff
@@ -0,0 +1,54 @@
+        @Getter
+        @AllArgsConstructor
+        @Builder
+        public static class Create {
```
사실상 Create, Update는 같은 내용이 아닌지요?

일반적으론,
```java
public class PostDTO {.. }

public class CreatePostRequest {
  public Post PostDTO;
 public string password
} 

```
...

로 작성합니다.

lecileJJL replied:
반영했습니다.

sesang06 commented:
> 확장성을 생각해 inner class 형식으로 작성했습니다.
왜 inner class를 사용하면 확장성이 생기는지요?

redhotkimchi commented:
> > 확장성을 생각해 inner class 형식으로 작성했습니다.
> 
> 왜 inner class를 사용하면 확장성이 생기는지요?
필요한 요청 마다 dto class를 선언하게 될 경우 dto 파일이 상황 별로 굉장히 많아지게 되어 관리하기 어려워지기 때문입니다. inner class를 사용할 경우 post관련 요청에 관하여 하나의 파일 안에서 관리 할 수 있게 됩니다.

lecileJJL committed:
Update PostDTO.java

### PR Type
<!— Please check the one that applies to this PR using "x". —>

- [ ] 버그수정(Bugfix)
- [ ] 기능개발(Feature)
- [ ] 코드 스타일 변경(Code style update) (formatting, local variables)
- [o] 리팩토링 (no functional changes, no api changes)
- [ ] 빌드 관련 변경
- [ ] 문서 내용 변경
- [ ] Other… Please describe:

### 요약(Summary)
postDTO.java update

### 상세 내용(Describe your changes)
postDTO.reqeust.create postDTO.reqeust.update
두 클라스를 병합했습니다.

### Issue Number or Link

sesang06 commented on src/main/java/com/example/PetnityServer/data/dto/PostDTO.java:
```diff
@@ -0,0 +1,44 @@
+        @Getter
+        @AllArgsConstructor
+        @Builder
+        public static class CreateAndUpdate {
```
디자인 도큐먼트엔 
```proto
message CreatePostRequest {
  Post post = 1;
  string password = 2;
}

message CreatePostResponse {
   // id , created_at, updated_at.
   Post post = 1;  
}

```
로 되어있는데, 디자인 도큐먼트의 설계와 일치하지 않습니다.

sesang06 commented on src/main/java/com/example/PetnityServer/data/dto/PostDTO.java:
```diff
@@ -0,0 +1,44 @@
+            @Getter
+            @AllArgsConstructor
+            @Builder
+            public static class ById {
+                private Long id;
+            }
```
ById? 어떤 건 CreateAndUpdate이고 어떤 건 ById... 네이밍에 일관성이 없습니다.

sesang06 commented on src/main/java/com/example/PetnityServer/data/dto/PostDTO.java:
```diff
@@ -0,0 +1,44 @@
+        @Getter
+        @AllArgsConstructor
+        @Builder
+        public static class Read {
```
CreateAndUpdate 와 Read 에서 사실상 같은 프로퍼티들을 묶어 두었는데요..
이 부분을 PostDTO 클래스로 묶어서 코드의 중복을 줄이라는 말이었습니다.

lecileJJL replied:
PostDTO 클래스로 묶어서 재 작성 하였습니다.

sesang06 commented on src/main/java/com/example/PetnityServer/data/dto/PostDTO.java:
```diff
@@ -0,0 +1,44 @@
+            private String message;
+            private Short dogAge;
+            private String dogName;
+            private String location;
```
Post DTO 를 디자인 도큐먼트에서
```proto


message Post {
   int32 id = 1;
   string title = 2;
   string nickname = 3;
   string dogName = 4;
   int32 dogAge = 5;
   string location = 6;
   string message = 7;
   repeated string imageURLs = 8;
   google.protobuf.Timestamp createdAt = 9;
   google.protobuf.Timestamp updatedAt = 10;
}
```
로 정했는데... createdAt과 updatedAt에 해당하는 프로퍼티를 추가해 주세요.

lecileJJL replied:
createdAt, updateAt 를 java.time DateTime 자료형으로 일시적으로 추가 했습니다.

sesang06 commented:
타이틀도 PR의 내용을 알 수 있도록 변경 부탁드립니다.

sesang06 commented:
> PostDTO 파일을 업로드 했습니다.

이렇게 막연하게 적지 마시고...PostDTO 가 무슨 역할을 하는지 자세히 기술해 주세요.

lecileJJL committed:
PostDTO 구조 수정

전체적인 구조 수정 및 데이터 베이스에 누락된 컬럼 추가

sesang06 commented on src/main/java/com/example/PetnityServer/data/dto/postDTO/ResponsePostDTO.java:
```diff
@@ -0,0 +1,10 @@
+
+@Getter
+@Builder
+public class ResponsePostDTO {
```
CreatePostRequestDTO 에 대응하는 네이밍이라면 CreatePostResponseDTO 가 맞다고 생각합니다.

lecileJJL replied:
반영했습니다

sesang06 commented on src/main/java/com/example/PetnityServer/data/dto/postDTO/ListPostRequestDTO.java:
```diff
@@ -0,0 +1,12 @@
+
+@Getter
+@Builder
+public class ListPostRequestDTO {
```
`ListPostsRequestDTO`가 단/복수에 맞다고 생각합니다.

lecileJJL replied:
반영했습니다.

sesang06 commented on src/main/java/com/example/PetnityServer/data/dto/postDTO/ListPostRequestDTO.java:
```diff
@@ -0,0 +1,12 @@
+public class ListPostRequestDTO {
+    private int pageCount;
+    private int pageSize ;
+    private String Keyword;
```
Keyword를 keyword로 바꿔 주세요.

lecileJJL replied:
반영했습니다.

sesang06 commented on src/main/java/com/example/PetnityServer/data/dto/postDTO/ListPostResponseDTO.java:
```diff
@@ -0,0 +1,12 @@
+@Getter
+@Builder
+public class ListPostResponseDTO {
+    private List<PostDTO> postDTOList;
```
`postDTOList`? 리스펀스를 래핑한 건데 외부에 `DTO`라는 명세를 내보낼 필요가 없어 보입니다.  `posts`면 됩니다.

lecileJJL replied:
반영했습니다.

sesang06 commented on src/main/java/com/example/PetnityServer/data/dto/postDTO/CreatePostRequestDTO.java:
```diff
@@ -0,0 +1,11 @@
+@Getter
+@Builder
+public class CreatePostRequestDTO {
+    PostDTO postDTO;
```
여기도 마찬가지입니다. PostDTO post로 바꿔 주세요.

lecileJJL replied:
반영했습니다.

sesang06 commented on build.gradle:
```diff
@@ -31,6 +31,7 @@ dependencies {
 	testRuntimeOnly 'org.junit.platform:junit-platform-launcher'
 	implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
 	implementation 'mysql:mysql-connector-java:8.0.33'
+	implementation 'com.google.protobuf:protobuf-java:3.17.0'
```
protobuf을 추가한 변경사항이 있나요?

lecileJJL replied:
document에 createdAt, updatedAt에 google.protobuf 자료형을 사용하여 반영하려 하였으나, 현재는 java.time을 사용하여 반영되지 않았습니다.

lecileJJL committed:
변수명 피드백 반영 했습니다.

변수명 일부 변경

lecileJJL committed:
변수명 재수정

변수명 재수정

sesang06 reviewed (approved):
LGTM
